examples/nbconverted/training_with_logging_example.py

```diff
@@ -0,0 +1,288 @@
+    axes[i, 1].set_title(f'Sample {i} - DNA')
+    axes[i, 1].axis('off')
+
+plt.tight_layout()
+plt.show()
+
+
+# ## Configure and train
+
+# In[ ]:
+
+
+# Hyperparameters
+batch_size = 16
+epochs = 100
+learning_rate = 0.001 # larger learning rate for demo purposes
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+# Batch with DataLoader
+train_loader = DataLoader(dataset, batch_size=batch_size, shuffle=True)
+
+# Model & Optimizer
+fully_conv_unet = UNet(
+    in_channels=1,
+    out_channels=1,
+    depth=4,
+    encoder_down_block='conv',
+    decoder_up_block='convt',
+    act_type='sigmoid'
+)
+optimizer = torch.optim.Adam(fully_conv_unet.parameters(), lr=learning_rate)
+
+# Plotting callback to visualize predictions during training
+plot_callback = PlotPredictionCallback(
+    name="plot_callback_with_train_data",
+    dataset=dataset,
+    indices=[0,1,2,3,4], # first 5 samples
+    plot_metrics=[torch.nn.L1Loss()],
+    every_n_epochs=5,
+    show_plot=False
+)
+
+# MLflow Logger
+logger = MlflowLogger(
+    name="logger",
+    experiment_name="vsf_examples",
+    tracking_uri="http://127.0.0.1:5000",
+    run_name="experiment_training_with_plots",
+    description="Training a UNet model on a simple dataset for demo purposes",
+    callbacks=[plot_callback],
+    save_model_at_train_end=True,
+    save_model_every_n_epochs=1,
+    save_best_model=True
+)
+
+# Initialize Trainer and start training
+trainer = SingleGeneratorTrainer(
+    model=fully_conv_unet,
+    optimizer=optimizer,
+    losses=[
+        torch.nn.L1Loss(),
+        MultiScaleStructuralSimilarityIndexMeasure(
+            data_range=1.0,
+            kernel_size=11,
+            sigma=1.5,
+        )
+    ],
+    loss_weights=[1.0, -1.0], # minimize L1 distance and maximize MS-SSIM
+    device=device,
+    train_loader=train_loader,
+    val_loader=None,
+    test_loader=None,
+)
+
+trainer.train(logger=logger, epochs=epochs)
+
+
+# ## Visualize training outcome through MLflow client
+
+# ### Display the last logged prediction plot artifact
+
+# In[ ]:
+
+
+# Create MLflow client
+client = MlflowClient(tracking_uri="http://127.0.0.1:5000")
+
+# Get the experiment by name
+experiment = client.get_experiment_by_name("vsf_examples")
+
+if experiment is None:
+    print("Experiment 'vsf_examples' not found")
+else:
+    # Search for runs with the specific run name
+    runs = client.search_runs(
+        experiment_ids=[experiment.experiment_id],
+        filter_string="tags.mlflow.runName = 'experiment_training_with_plots'"
+    )
+
+    if len(runs) == 0:
+        print("No runs found with name 'experiment_training_with_plots'")
+    else:
+        # Get the most recent run (first in list)
+        run = runs[0]
+        print(f"Run ID: {run.info.run_id}")
+        print(f"Run Name: {run.data.tags.get('mlflow.runName')}")
+
+        plot_artifacts = client.list_artifacts(run.info.run_id, path='plots/epoch/plot_predictions/')
```

Comment:
Recommend adding a code comment here about what "artifacts" means in this context.